Source: Backported from texinfo.tex in TeXLive 2007
Upstream: Affected branches are long dead; newer ones are fixed
Reason: If the etex binary is pdftex (as in TeXLive), don't force it
        into PDF mode when texi2dvi expects DVI output.  This affects
        the test cases, and possibly also anyone generating a
        distribution tarball of project that uses an affected automake
        version and that has texinfo documentation.

--- automake-1.7.9/lib/texinfo.tex~	2003-11-04 18:53:52.000000000 +0000
+++ automake-1.7.9/lib/texinfo.tex	2008-06-18 21:46:07.000000000 +0100
@@ -1064,16 +1064,21 @@
 \newif\ifpdf
 \newif\ifpdfmakepagedest
 
+% when pdftex is run in dvi mode, \pdfoutput is defined (so \pdfoutput=1
+% can be set).  So we test for \relax and 0 as well as \undefined,
+% borrowed from ifpdf.sty.
 \ifx\pdfoutput\undefined
-  \pdffalse
-  \let\pdfmkdest = \gobble
-  \let\pdfurl = \gobble
-  \let\endlink = \relax
-  \let\linkcolor = \relax
-  \let\pdfmakeoutlines = \relax
 \else
-  \pdftrue
-  \pdfoutput = 1
+  \ifx\pdfoutput\relax
+  \else
+    \ifcase\pdfoutput
+    \else
+      \pdftrue
+    \fi
+  \fi
+\fi
+
+\ifpdf
   \input pdfcolor
   \pdfcatalog{/PageMode /UseOutlines}%
   \def\dopdfimage#1#2#3{%
@@ -1254,7 +1259,13 @@
     \startlink attr{/Border [0 0 0]} goto name{\pdfmkpgn{#1}}
     \linkcolor #1\endlink}
   \def\done{\edef\st{\global\noexpand\toksA={\the\toksB}}\st}
-\fi % \ifx\pdfoutput
+\else
+  \let\pdfmkdest = \gobble
+  \let\pdfurl = \gobble
+  \let\endlink = \relax
+  \let\linkcolor = \relax
+  \let\pdfmakeoutlines = \relax
+\fi % \ifpdf
 
 
 \message{fonts,}
